

# EDA on McDonald's Data



## Table of Contents

1. [Problem Statement](#section1)<br>
2. [Importing Packages](#section2)<br>
3. [Loading Data](#section3)<br>
  - 3.1 [Description of the Datasets](#section301)<br>
  - 3.2 [Pandas Profiling before Data Preprocessing](#section302)<br>
4. [Data Preprocessing](#section4)<br>
  - 4.1 [Data Preprocessing](#section401)<br>
  - 4.2 [Pandas Profiling after Data Preprocessing](#section402)<br>
5. [Exploratory Data Analysis](#section5)<br>
  - 5.1 [**Analysis based on Outlet metrics**](#section501)<br>
    - 5.1.1 [How many stores are actually owned and run by McDonald's?](#section50101)<br>
    - 5.1.2 [Which outlets generate maximum revenue for the company?](#section50102)<br>
    - 5.1.3 [Top 10 Outlets in terms of Profit](#section50103)<br>
    - 5.1.4 [Outlets with highest Number of Employees](#section50104)<br>
    - 5.1.5 [Where do you find the maximum footfall in McDonald's?](#section50105)<br><br>
  - 5.2 [**Analysis based on Nutritional Value**](#section502)<br>
      - 5.2.1 [Which is the most common category on menu?](#section50201)<br>
      - 5.2.2 [How many calories (energy) on an average are present in each category of McDonald's menu?](#section50202)<br>
      - 5.2.3 [Is there any difference in the nutritional value of a grilled vs crispy chicken?](#section50203)<br>
      - 5.2.4 [On an average, how much sugar consumption happens when you order Desserts, Beverages, Smoothies and Shakes?](#section50204)<br>
      - 5.2.5 [In what sort of foods and beverages do manufacturers include fiber?](#section50205)<br> 
      - 5.2.6 [How to select Nutritious and Non-Nutritious food from menu?](#section50206)<br> 
      - 5.2.7 [Which is the most proteinaceous item on the menu?](#section50207)<br> <br>
  - 5.3 [**Analysis based on Geographical Information**](#section503)<br>
    - 5.3.1 [Which city has the highest number of McDonald's outlets per capita?](#section50301)<br>
    - 5.3.2 [Which state has the highest number of McDonald's outlets per capita?](#section50302)<br>
    - 5.3.3 [Outlet locations in India](#section50303)<br>
    - 5.3.4 [Outlet locations in India](#section50304)<br><br>
  - 5.4 [**Analysis based on Outlet metrics and Nutritional value**](#section504)<br>
    - 5.4.1 [Which outlets have most nutritious item as their best selling item?](#section50401)<br>
    - 5.4.2 [What is the Revenue of the outlet based on the category of its best selling item?](#section50402)<br>
    - 5.4.3 [How much is the Total Selling Price of the outlet based on the category of its best selling item?](#section50403)<br>
    - 5.4.4 [How much is the Total Cost Price of the outlet based on the category of its best selling item?](#section50404)<br><br>
  - 5.5 [**Analysis based on Outlet Metrics and Geographical Information**](#section505)<br>
    - 5.5.1 [Where did McDonald's entered into Joint Venture to start their Outlets?](#section50501)<br>
    - 5.5.2 [Top 10 outlets in India and US based on revenue](#section50502)<br>
    - 5.5.3 [What is the mean revenue of outlets in each US state?](#section50503)<br>
    - 5.5.4 [How much is the Gross Profit Margin of each outlet?](#section50504)<br><br>
  - 5.6 [**Analysis based on Nutritional value and Geographical information**](#section506)<br>
    - 5.6.1 [Comparison of Nutritional content of each category between India and US](#section50601)<br>
    - 5.6.2 [How are Indian menu items compare to US menu items in terms of nutrition?](#section50602)<br><br>
  - 5.7 [**Analysis based on outlet metrics, menu items, and geographical information**](#section507)<br>
    - 5.7.1 [How is the revenue generation based on different menu items in India and US?](#section50701)<br><br>
6. [Conclusion](#section6)
  - 6.1 [Actionable Insights](#section601)

<a id=section1></a>
## 1. Problem Statement

**McDonald's** is hip all over the world, including in India. However, the McDonald's menu is not universal. In **India**, there are some **menu items** that are similar, but others that are much **different from the U.S. menu**. Through **comparing the McDonald's menu in the U.S. and India**, we can see that fast food menus can truly **give us a perspective on cultural and economic similarities and differences throughout the world**. 



---

<a id=section2></a>
## 2. Importing Packages

In [12]:
import numpy as np                     

import pandas as pd
pd.set_option('mode.chained_assignment', None)      # To suppress pandas warnings.
pd.set_option('display.max_colwidth', -1)           # To display all the data in each column
pd.options.display.max_columns = 50                 # To display every column of the dataset in head()

import warnings
warnings.filterwarnings('ignore')                   # To suppress all the warnings in the notebook.

In [13]:
import matplotlib.pyplot as plt
%matplotlib inline

import seaborn as sns
sns.set(style='whitegrid', font_scale=1.3, color_codes=True)      # To apply seaborn styles to the plots.

In [14]:
# Install and update plotly using this command to the latest version (higher than version 4.0.0)

!pip install plotly --upgrade

In [15]:
# Install chart_studio, please use this command. 

!pip install chart-studio

In [16]:
# Making plotly specific imports
# These imports are necessary to use plotly offline without signing in to their website.

from plotly.offline import init_notebook_mode, iplot
import plotly.graph_objs as go
import chart_studio.plotly as py
from plotly import tools
init_notebook_mode(connected=True)

---

### Disclaimer: The images are only for illustration purpose. Please don't compare them with the data.

<a id=section3></a>
## 3. Loading Data

In this **Data Visualization** sheet we are using two datasets about **McDonald's**. 

<img src="https://raw.githubusercontent.com/insaid2018/Term-1/master/Images/mcdonalds_logo.jpg" width=750 height=750><br/>

- 1. **Nutrition Facts for McDonald's Menu** dataset: This dataset contains information about the **nutritional content** present in different items on **McDonald's Menu** for **India and US** stores.


- 2. **McDonald's Outlet's Information** dataset: This dataset contains information about various **McDonald's outlets** in **India and US**. 

### Importing 1st Dataset

In [28]:
# Importing the 1st dataset (Nutrition Facts for McDonald's Menu dataset).

df_menu = pd.read_excel('https://github.com/amity1415/DS/raw/main/EKeeda/Rawdata/mcdonalds_menu.xlsx')
df_menu.head()

,Category,Item,Serve_Size,Energy,Protein,Total_Fat,Saturated_Fat,Trans_Fat,Cholestrol,Carbohydrates,Sugars,Dietary_Fibre,Sodium
0,Breakfast,Egg & Cheese Muffin,112.0,290.0,14.0,13.0,7.0,0.2,244.0,28.0,2.0,2.0,620.0
1,Breakfast,Sausage McMuffm,112.0,273.0,16.0,11.0,5.7,0.2,50.0,28.0,2.0,2.0,950.0
2,Breakfast,Sausage & Egg McMuffm,157.0,355.0,22.0,17.0,7.9,0.2,277.0,29.0,2.0,2.0,1020.0
3,Breakfast,Veg McMuffm,119.0,299.0,10.0,12.0,7.2,0.2,21.0,38.0,3.0,3.0,1000.0
4,Breakfast,Veg Supreme Muffin,139.0,299.0,7.0,13.0,5.2,0.2,11.0,39.0,4.0,4.0,960.0


<a id=section301></a>
### Description of the Dataset

- This dataset provides a **nutrition analysis** of different menu items on the **McDonald's menu** in **India and US**, including breakfast, chicken wings, shake, snacks, hot beverages, cold beverages and desserts.


- Detailed information about each nutrient column can be found at this [link](https://www.fda.gov/food/nutrition-education-resources-materials/how-understand-and-use-nutrition-facts-label).


| Column Name                    | Description                                                                                |
| -------------------------------|:------------------------------------------------------------------------------------------:| 
| Category                       | The category to which an item belong to.                                                   | 
| Item                           | Name of the item.                                                                          | 
| Serve_Size                     | The weight of a single serving (g).                                                        | 
| Energy                         | Number of calories in the item (kcal).                                                     |
| Protein                        | Protein in the item (g).                                                                   |
| Total_Fat                      | Total Fat in the item (g).                                                                 |
| Saturated_Fat                  | Saturated Fat in the item (g).                                                             |
| Trans_Fat                      | Trans Fat in the item (g).                                                                 |
| Cholestrol                     | Cholestrol in the item (mg).                                                               |
| Carbohydrates                  | Carbohydrates in the item (g).                                                             |
| Sugars                         | Sugars in the item (g).                                                                    |
| Dietary_Fibre                  | Dietary Fibres in the item (g).                                                            |
| Sodium                         | Sodium in the item (mg).                                                                   |


<img src="https://raw.githubusercontent.com/amity1415/DS/main/EKeeda/Images/EDA/Mcd_EDA1.png" width=750 height=750><br/>


In [7]:
df_menu.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 340 entries, 0 to 339
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Category       340 non-null    object 
 1   Item           340 non-null    object 
 2   Serve_Size     340 non-null    object 
 3   Energy         340 non-null    float64
 4   Protein        340 non-null    float64
 5   Total_Fat      340 non-null    float64
 6   Saturated_Fat  340 non-null    float64
 7   Trans_Fat      340 non-null    float64
 8   Cholestrol     340 non-null    float64
 9   Carbohydrates  340 non-null    float64
 10  Sugars         340 non-null    float64
 11  Dietary_Fibre  340 non-null    float64
 12  Sodium         340 non-null    object 
dtypes: float64(9), object(4)
memory usage: 34.7+ KB


- ```info``` function gives us the following insights into the df_menu dataframe:

  - There are a total of **340 samples (rows)** and **13 columns** in the dataframe.
  
  - There are **9 columns** with a **numeric** datatype and **4 columns** with an **object** datatype.
  
  - There are **0 missing** values in the data.

### Importing 2nd Dataset

In [29]:
# Importing the 2nd dataset (McDonald's Store Information dataset).

df_store = pd.read_excel('https://github.com/amity1415/DS/raw/main/EKeeda/Rawdata/mcdonalds_outlets.xlsx')
df_store.head()

,Brand,Store_ID,Store_Name,Ownership_Type,Street_Address,City,State,Country,Postcode,Phone_Number,Timezone,Longitude,Latitude,Revenue,Profits,Selling_Price,Cost_Price,Gross_Profit_Margin,Number_of_Employees,Customers,Best_Selling_Item
0,McDonald's,23149-228271,Banjara Hills,Joint Venture,"Lower Ground Floor, GVK One, Road Number 1, Banjara Hills",Hyderabad,AP,IN,500034,NaN,GMT+05:30 Asia/New_Delhi,78.45,17.42,2.117344,0.171584,1.655165,0.907433,0.747732,34.311197,3979.583117,Egg & Cheese Muffin
1,McDonald's,23191-228548,Kukatpally,Joint Venture,"Upper Ground Floor, Forum Sujana Mall, Kukatpally",Hyderabad,AP,IN,500072,NaN,GMT+05:30 Asia/New_Delhi,78.39,17.48,1.058504,0.054645,0.895944,0.453645,0.442299,25.487533,1156.010620,Sausage McMuffm
2,McDonald's,23193-228546,Madhapur,Joint Venture,"Lower Ground Floor, Inorbit Mall, Madhapur",Hyderabad,AP,IN,500081,NaN,GMT+05:30 Asia/New_Delhi,78.39,17.43,4.505020,0.663867,2.864311,1.930723,0.933588,54.208502,10346.720786,Sausage & Egg McMuffm
3,McDonald's,23180-228545,Jubilee Hills,Joint Venture,"Ground Floor, Road No. 92, Near Apollo hospital, Jubilee Hills",Hyderabad,AP,IN,500033,NaN,GMT+05:30 Asia/New_Delhi,78.42,17.42,3.405530,0.397835,2.394072,1.459513,0.934559,45.046083,7414.746544,Veg McMuffm
4,McDonald's,24457-238129,Hi-Tech City,Joint Venture,"Upper Ground Floor, Phoenix tower A, Opposite Trident Hotel, Madhapur Village",Hyderabad,AP,IN,500084,NaN,GMT+05:30 Asia/New_Delhi,78.38,17.45,7.332713,1.656524,3.395053,3.142591,0.252461,77.772607,17887.234107,Veg Supreme Muffin


### Description of the Dataset

- This dataset provides information different **McDonald's India outlets** like their Store ID, Address, City, State, Revenue, Profits, Number of Employees, Most Selling Item, etc.


| Column Name                    | Description                                                                                |
| -------------------------------|:------------------------------------------------------------------------------------------:| 
| Brand                          | Brand name.                                                                                | 
| Store ID                       | ID of the store.                                                                           | 
| Store Name                     | Name of the store.                                                                         |
| Ownership Type                 | Ownership type of the store.                                                               |
| Street Address                 | Address of the store's location.                                                           |
| City                           | City of the store's location.                                                              |
| State                          | State of the store's location.                                                             |
| Country                        | Country of the store's location.                                                           |
| Postcode                       | Postcode of the store's location.                                                          |
| Phone Number                   | Phone number of the store.                                                                 |
| Timezone                       | Timezone of the store's location.                                                          |
| Longitude                      | Longitude of the store's location.                                                         |
| Latitude                       | Latitude of the store's location.                                                          |
| Revenue                        | Revenue of the store (in million Indian Rupees).                                           |
| Profits                        | Profits of the store (in million Indian Rupees).                                           |
| Selling Price                  | Total Selling Price of items of the store (in million Indian Rupees).                      |
| Cost Price                     | Total Cost Price of items of the store (in million Indian Rupees).                         |
| Gross Profit Margin            | Gross Profit Margin of the store (in million Indian Rupees).                               |
| Number of Employees            | Number of employees of the store.                                                          |
| Customers                      | Number of monthly customeres of the store.                                                 |
| Best Selling Item              | Most selling item of the store.                                                            |


<img src="https://raw.githubusercontent.com/amity1415/DS/main/EKeeda/Images/EDA/Mcd_EDA2.png" width=750 height=750><br/>


In [9]:
df_store.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 340 entries, 0 to 339
Data columns (total 21 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Brand                340 non-null    object 
 1   Store_ID             340 non-null    object 
 2   Store_Name           340 non-null    object 
 3   Ownership_Type       340 non-null    object 
 4   Street_Address       340 non-null    object 
 5   City                 340 non-null    object 
 6   State                340 non-null    object 
 7   Country              340 non-null    object 
 8   Postcode             340 non-null    int64  
 9   Phone_Number         248 non-null    object 
 10  Timezone             340 non-null    object 
 11  Longitude            340 non-null    float64
 12  Latitude             340 non-null    float64
 13  Revenue              340 non-null    float64
 14  Profits              340 non-null    float64
 15  Selling_Price        340 non-null    flo

- ```info``` function gives us the following insights into the df_store dataframe:

  - There are a total of **340 samples (rows)** and **21 columns** in the dataframe.
  
  - There are **10 columns** with a **numeric** datatype and **11 columns** with an **object** datatype.
  
  - There are **missing** values in the **Phone Number** column.

- We are going to perform a **merge** on the **two dataframes**, **df_store and df_menu** on the basis of **Best Selling Item** of **df_store** dataframe and **Item** column of **df_menu** dataframe to create a new dataframe **df_merge**, in order to make our **EDA** easier.

In [30]:
df_merge = pd.merge(df_store, df_menu, how='inner', left_on='Best_Selling_Item', right_on='Item').drop(['Item'], 1)
df_merge.head()

,Brand,Store_ID,Store_Name,Ownership_Type,Street_Address,City,State,Country,Postcode,Phone_Number,Timezone,Longitude,Latitude,Revenue,Profits,Selling_Price,Cost_Price,Gross_Profit_Margin,Number_of_Employees,Customers,Best_Selling_Item,Category,Serve_Size,Energy,Protein,Total_Fat,Saturated_Fat,Trans_Fat,Cholestrol,Carbohydrates,Sugars,Dietary_Fibre,Sodium
0,McDonald's,23149-228271,Banjara Hills,Joint Venture,"Lower Ground Floor, GVK One, Road Number 1, Banjara Hills",Hyderabad,AP,IN,500034,NaN,GMT+05:30 Asia/New_Delhi,78.45,17.42,2.117344,0.171584,1.655165,0.907433,0.747732,34.311197,3979.583117,Egg & Cheese Muffin,Breakfast,112.0,290.0,14.0,13.0,7.0,0.2,244.0,28.0,2.0,2.0,620.0
1,McDonald's,23191-228548,Kukatpally,Joint Venture,"Upper Ground Floor, Forum Sujana Mall, Kukatpally",Hyderabad,AP,IN,500072,NaN,GMT+05:30 Asia/New_Delhi,78.39,17.48,1.058504,0.054645,0.895944,0.453645,0.442299,25.487533,1156.010620,Sausage McMuffm,Breakfast,112.0,273.0,16.0,11.0,5.7,0.2,50.0,28.0,2.0,2.0,950.0
2,McDonald's,23193-228546,Madhapur,Joint Venture,"Lower Ground Floor, Inorbit Mall, Madhapur",Hyderabad,AP,IN,500081,NaN,GMT+05:30 Asia/New_Delhi,78.39,17.43,4.505020,0.663867,2.864311,1.930723,0.933588,54.208502,10346.720786,Sausage & Egg McMuffm,Breakfast,157.0,355.0,22.0,17.0,7.9,0.2,277.0,29.0,2.0,2.0,1020.0
3,McDonald's,23180-228545,Jubilee Hills,Joint Venture,"Ground Floor, Road No. 92, Near Apollo hospital, Jubilee Hills",Hyderabad,AP,IN,500033,NaN,GMT+05:30 Asia/New_Delhi,78.42,17.42,3.405530,0.397835,2.394072,1.459513,0.934559,45.046083,7414.746544,Veg McMuffm,Breakfast,119.0,299.0,10.0,12.0,7.2,0.2,21.0,38.0,3.0,3.0,1000.0
4,McDonald's,24457-238129,Hi-Tech City,Joint Venture,"Upper Ground Floor, Phoenix tower A, Opposite Trident Hotel, Madhapur Village",Hyderabad,AP,IN,500084,NaN,GMT+05:30 Asia/New_Delhi,78.38,17.45,7.332713,1.656524,3.395053,3.142591,0.252461,77.772607,17887.234107,Veg Supreme Muffin,Breakfast,139.0,299.0,7.0,13.0,5.2,0.2,11.0,39.0,4.0,4.0,960.0


In [ ]:
df_merge.info()

- ```info``` function gives us the following insights into the df_merge dataframe:

  - There are a total of **340 samples (rows)** and **33 columns** in the dataframe.
  
  - There are **19 columns** with a **numeric** datatype and **14 columns** with an **object** datatype.
  
  - There are **missing** values in the **Phone Number** column.

In [ ]:
df_merge.describe()

- ```describe``` function gives us the following insights into the data:
  
  - **Revenue** column is having an almost **normal distribution** because the **mean (25.8534)** value is close to the **median (27.0561)** value.

  - The distribution of **Energy** is **positive (right) skewed** because the **mean** of **Energy (337.800)** lies to the **right** of the **median(50%)** value **(299.500)**, on a **distribution plot**. 
  
  - The **max** value of **Energy (1880.000)** is an **outlier** because it lie far away from the mean.
  
  - Similarly comparing the **75%** and **max** values of the other columns we can see that there are **outliers** present in some other columns.

---

<a id=section302></a>
### 3.2 Pandas Profiling before Data Preprocessing

- Here, we will perform **Pandas Profiling before preprocessing** our dataset, so we will name the **output file** as __profiling_before_preprocessing.html__. 


- The file will be stored in the directory of your notebook. Open it using the jupyter notebook file explorer and take a look at it and see what insights you can develop from it. 


- Or you can **output the profiling report** in the **current jupyter notebook** as well as shown in the code below. 

In [ ]:
# To install pandas profiling please run this command.

!pip install pandas-profiling --upgrade

In [ ]:
import pandas_profiling 

In [ ]:
# To output pandas profiling report to an external html file.
# Saving the output as profiling_before_preprocessing.html
'''
profile = df_merge.profile_report(title='Pandas Profiling before Data Preprocessing')
profile.to_file(output_file="profiling_before_preprocessing.html")
'''

# To output the pandas profiling report on the notebook.

df_merge.profile_report(title='Pandas Profiling before Data Preprocessing', style={'full_width':True})

**Observations from Pandas Profiling before Data Processing** <br><br>
__Dataset info__:
- Number of variables: 33
- Number of observations: 340
- Missing cells: 92 (0.8%)


__Variables types__: 
- Numeric = 16
- Categorical = 10
- Text (Unique) = 3
- Rejected = 4

<br>

- **Brand** has **constant value "McDonald's"**.


- **Phone_Number** has **92 (27.1%) missing values**.


- **Number_of_Employees** is **highly correlated** with **Cost_Price (ρ = 0.9091375321)**.


- **Revenue** is **highly correlated** with **Number_of_Employees (ρ = 0.9091375321)**.


- **Total_Fat** is **highly correlated** with **Energy (ρ = 0.9065)**.


- **City** has a **high cardinality (139 distinct values)**.


- **Phone_Number** has a **high cardinality (249 distinct values)**.


- **Serve_Size** has a **high cardinality (161 distinct values)**.


- **Street_Address** has a **high cardinality (339 distinct values)**.




- **Cholestrol** has a **skewness** of **3.8984** and **kurtosis** of **18.847**.

  - The distribution is **positive (right) skewed** as the **mean (47.544)** is to the **right** of the **median (25)**.
  
  - **95%** samples have a **Cholestrol** less than **250**.
  
  - The high value of **kurtosis** implies that the tail is heavy and there is a presence of outliers in the data (**maximum value = 575 is an outlier**).
  

- **Carbohydrates** has a **skewness** of **0.9975** and **kurtosis** of **1.8172**.

  - **Maximum value = 141** is an **outlier**, as it lies far away from the **Q3 (75th percentile)** value **(55)**.


- **Category** has data divided into **14** distinct values.
  
  - __Hot Beverages__ has the **highest** share with **99** samples **(29.1% share)** and then **Breakfast** has **49** samples **(14.4% share)**.
 			
  
- **Country** has data divided into **2** distinct values.
  
  - __US__ has **258** samples **(75.9% share)** and **IN** has **82** samples **(24.1% share)**.
  
 
- **Energy** has a **skewness** of **1.4793** and **kurtosis** of **5.7022**.

  - **Maximum value = 1880** is an **outlier**, as it lies far away from the **Q3 (75th percentile)** value **(460)**.
 

- **Ownership_Type** has data divided into **3** distinct categories.
  
  - __Company Owned__ has **137** samples **(40.3% share)**, **Licensed** has **121** samples **(35.6% share)**, and **Joint Venture** has **82** samples **(24.1% share)**.
  
  
- **Protein** has a **skewness** of **1.639** and **kurtosis** of **6.0817**.

  - **Maximum value = 87** is an **outlier**, as it lies far away from the **Q3 (75th percentile)** value **(17)**.
  
  
- All the **remaining columns** can be studied in the same way.

---

<a id=section4></a>
## 4. Data Preprocessing

<a id=section401></a>
### 4.1 Data Preprocessing

- Here we will perform **Data Preprocessing** on our dataset, to make the data usable for **EDA**.


- In order to avoid any errors, please run **Pandas Profiling** before running the code below.

In [ ]:
df_merge.head()

- Since the column **Brand** has a **constant value** for all the samples, we will drop this column.


- Also the columns **Postcode, Phone Number, and Timezone** will be **dropped** because they **don't provide** any additional **insights** into the data.

In [ ]:
df_merge.drop(['Brand', 'Postcode', 'Phone_Number', 'Timezone'], 1, inplace=True)

In [ ]:
df_merge.head()

- **Rounding** the **values** of **Revenue, Profits, Sellling_Price, Cost_Price, Gross_Profit_Margin** columns to 3 decimal places.

In [ ]:
df_merge = df_merge.round(decimals=3)
df_merge.head()

- The **Number of Employees** and **Customers** columns have values in **decimals**, which is not possible because **count of people** is an example of **natural number**.


- This looks like a **Data Entry error**, and we will modify the values to lie in the **set of natural numbers**.

In [ ]:
## Using apply(np.floor) to get the floor of the values.

df_merge['Number_of_Employees'] = df_merge['Number_of_Employees'].apply(np.floor).astype(int)
df_merge['Customers'] = df_merge['Customers'].apply(np.floor).astype(int)
df_merge.head()

- **Serve_Size** column has some **irregularities** present, let's fix them.  

In [ ]:
# Checking the different types of values in the Serve_Size column. Looking at every 4th value after 50th value.

df_merge['Serve_Size'].unique()[::5]

In [ ]:
# Removing all the unwanted text (oz, cookie, carton, cup, (, ), g, ml, etc.) from the Serve_Size column.
# Using '\d' in the str.replace which is used to specify all the digits between 0-9.

df_merge['Serve_Size'] = df_merge['Serve_Size'].astype(str).str.replace('[\.\d]+ oz ', '').str.replace('[\d]+ cookie ', '').str.replace('[\d]+ carton ', '').str.replace(' cup', '').str.replace('[\(\)]', '').str.replace(' g', '').str.replace(' ml', '').str.replace(' [\d]+', '')

In [ ]:
# We can see that only the samples having 'fl oz' in them remain in the column.

df_merge['Serve_Size'].unique()[110::3]

- Here we can see that some of the values in the **Serve_Size** column have their **weight** mention in **oz**.


- **1 ounce (oz)** is **equal** to **28.34952 grams (g)**.


- We will **extract** the **oz** values from these samples and **replace** them with their respective values in **grams**.

In [ ]:
# Only text to remove from the Serve_Size column is ' fl oz'. Saving the index of rows having these values into serve_index.

serve_index = df_merge[df_merge['Serve_Size'].str.contains(' fl oz')].index

In [ ]:
# Removing ' fl oz' from the rows and replacing it with the value in grams.

df_merge.iloc[serve_index, 18] = (df_merge.iloc[serve_index, 18].str.replace(' fl oz', '').astype(float) * 28.34952)

In [ ]:
df_merge['Serve_Size'] = df_merge['Serve_Size'].astype(int)
df_merge['Serve_Size'].unique()[::8]

- Lastly fixing the **Sodium** column.

In [ ]:
df_merge['Sodium'].value_counts()[:5]

In [ ]:
# Calculating the median of the rows which don't have Sodium value '-'.

df_merge[~(df_merge['Sodium'] == '-')]['Sodium'].median()

In [ ]:
df_merge['Sodium'] = df_merge['Sodium'].astype(str).str.replace('-', '230').astype(int)

In [ ]:
df_merge['Sodium'].value_counts()[:5]

In [ ]:
df_merge.isnull().sum()

---

<a id=section402></a>
### 4.2 Pandas Profiling after Data Preprocessing

- Here, we will perform **Pandas Profiling after preprocessing** our dataset, so we will name the **output file** as __profiling_after_preprocessing.html__.

In [ ]:
# To output pandas profiling report to an external html file.
# Saving the output as profiling_after_preprocessing.html
'''
profile = df_merge.profile_report(title='Pandas Profiling after Data Preprocessing')
profile.to_file(output_file="profiling_after_preprocessing.html")
'''

# To output the pandas profiling report on the notebook.

df_merge.profile_report(title='Pandas Profiling after Data Preprocessing', style={'full_width':True})

**Observations from Pandas Profiling before Data Processing** <br><br>
__Dataset info__:
- Number of variables: 29
- Number of observations: 340
- Missing cells: 0 (0.0%)


__Variables types__: 
- Numeric = 17
- Categorical = 6
- Text (Unique) = 3
- Rejected = 3

<br>

- **Sodium** has a **skewness** of **1.6021** and **kurtosis** of **3.3506**.

  - **Maximum value = 3600** is an **outlier**, as it lies far away from the **Q3 (75th percentile)** value **(790)**.

---

<a id=section5></a>
## 5. Exploratory Data Analysis

**Exploratory Data Analysis(EDA)** is an **approach to analyzing data sets** to **summarize** their **main characteristics**, often with **visual methods**. 


- It includes **cleaning, munging, combining, reshaping, slicing, dicing, and transforming data** for **analysis** purpose.


- The **primary goal of EDA** is to **maximize the analyst's insight into a data set** and into the **underlying structure of a data set**, while **providing all of the specific items** that an **analyst would want to extract from a data set**, such as:
 <br><br> 
  - A list of **outliers**.<br><br>
  
  - A **good-fitting model**.  <br><br>
  
  - **Estimates for parameters**.<br><br>
  
  - **Uncertainties** for those **estimates**.<br><br>
  
  - A **ranked list** of **important factors**.<br><br>
  
  - **Conclusions** as to whether **individual factors are statistically significant**.<br><br>
  
  - A **sense of robustness of conclusions**.<br><br>
  
  - **Optimal settings**.<br><br>

<a id=section501></a>
## 5.1 Analysis Based on Outlet Metrics

<a id=section50101></a>
### 5.1.1 How many stores are actually owned and run by McDonald's?

In [ ]:
df_merge['Ownership_Type'].value_counts()

In [ ]:
df_merge['Ownership_Type'].value_counts().plot(kind='pie', explode=[0.05,0.05,0.05], fontsize=14, autopct='%3.1f%%', 
                                               figsize=(10,10), shadow=True, startangle=135, legend=True, cmap='summer')

plt.ylabel('Ownership Type')
plt.title('Donut Plot showing the proportion of each Ownership_Type value')

- **40%** of the outlets are **company owned** and **35%** are **Licensed** outlets.


- Around **24%** of the outlets are on **Joint venture**. This is mostly in countries where 100% FDI is/was not allowed to start their own outlets.

<a id=section50102></a>
### 5.1.2 Which outlets generate maximum revenue for the company?

In [ ]:
df_merge.iloc[df_merge['Revenue'].sort_values(ascending=False)[:10].index, :]

In [ ]:
data = [go.Scattergeo(lon=df_merge.iloc[df_merge['Revenue'].sort_values(ascending=False)[:10].index, :]['Longitude'], 
                      lat=df_merge.iloc[df_merge['Revenue'].sort_values(ascending=False)[:10].index, :]['Latitude'], 
                      text=df_merge.iloc[df_merge['Revenue'].sort_values(ascending=False)[:10].index, :][['Store_Name', 'Revenue']], 
                      name='',
                      marker=dict(size=df_merge.iloc[df_merge['Revenue'].sort_values(ascending=False)[:10].index, :]['Revenue']/2, 
                                  color='Green'))]


In [ ]:
layout = go.Layout(title=go.layout.Title(text='Top 10 outlets in terms of Revenue (in million INR)'), 
                   showlegend=False,
                   geo=dict(scope='usa', projection=go.layout.geo.Projection(type='albers usa'), showlakes=True,
                            lakecolor='rgb(255, 255, 255)'))

In [ ]:
fig = go.Figure(data=data, layout=layout)

In [ ]:
iplot(fig)

- All the stores in our **Top 10** for **maximum revenue generation** lie in the **US**.


- The **size** of the **marker** gives the **Revenue** of the outlet, and it can be seen that all the outlets in the **Top 10** have very **similar revenue** (near about **48 - 49 million INR**). 


- The **tooltip** shows the **geographical coordinates, name** and **revenue** of the **outlet**.

<a id=section50103></a>
### 5.1.3 Top 10 Outlets in terms of Profit

In [ ]:
df_merge.iloc[df_merge['Profits'].sort_values(ascending=False)[:10].index, :]

In [ ]:
data = [go.Scattergeo(lon=df_merge.iloc[df_merge['Profits'].sort_values(ascending=False)[:10].index, :]['Longitude'], 
                      lat=df_merge.iloc[df_merge['Profits'].sort_values(ascending=False)[:10].index, :]['Latitude'], 
                      text=df_merge.iloc[df_merge['Profits'].sort_values(ascending=False)[:10].index, :][['Store_Name', 'Profits']], 
                      name='',
                      marker=dict(size=df_merge.iloc[df_merge['Profits'].sort_values(ascending=False)[:10].index, :]['Profits'], 
                                  color='Blue'))]


In [ ]:
layout = go.Layout(title=go.layout.Title(text='Top 10 outlets in terms of Profit (in million INR)'), 
                   showlegend=False,
                   geo=dict(scope='usa', projection=go.layout.geo.Projection(type='albers usa'), showlakes=True,
                            lakecolor='rgb(255, 255, 255)'))

In [ ]:
fig = go.Figure(data=data, layout=layout)

In [ ]:
iplot(fig)

- All the stores in our **Top 10** for **maximum profits** also lie in the **US**.


- The **size** of the **marker** gives the **Profit** of the outlet.


- The **tooltip** shows the **geographical coordinates, name** and **profits** of the **outlet**.

<a id=section50104></a>
### 5.1.4 Outlets with highest Number of Employees

In [ ]:
df_merge.iloc[df_merge['Number_of_Employees'].sort_values(ascending=False)[:10].index, :]

In [ ]:
data = [go.Scattergeo(lon=df_merge.iloc[df_merge['Number_of_Employees'].sort_values(ascending=False)[:10].index, :]['Longitude'], 
                      lat=df_merge.iloc[df_merge['Number_of_Employees'].sort_values(ascending=False)[:10].index, :]['Latitude'], 
                      text=df_merge.iloc[df_merge['Number_of_Employees'].sort_values(ascending=False)[:10].index, :][['Store_Name', 'Number_of_Employees']], 
                      name='',
                      marker=dict(size=df_merge.iloc[df_merge['Number_of_Employees'].sort_values(ascending=False)[:10].index, :]['Number_of_Employees']/7, 
                                  color='orange'))]


In [ ]:
layout = go.Layout(title=go.layout.Title(text='Top 10 outlets in terms of Number of Employees'), showlegend=False,
                   geo=dict(scope='usa', projection=go.layout.geo.Projection(type='albers usa'), showlakes=True,
                            lakecolor='rgb(255, 255, 255)'))

In [ ]:
fig = go.Figure(data=data, layout=layout)

In [ ]:
iplot(fig)

- All the stores in our **Top 10** for **highest number of employees** lie in the **US** as well.


- The **number of employees** in the **top 10** lie in the **range** between **145-149**.


- The **tooltip** shows the **geographical coordinates, name** and **number of employees** of the **outlet**.

<a id=section50105></a>
### 5.1.5 Where do you find the maximum footfall in McDonald's?

In [ ]:
df_merge.iloc[df_merge['Customers'].sort_values(ascending=False)[:10].index, :]

In [ ]:
data = [go.Scattergeo(lon=df_merge.iloc[df_merge['Customers'].sort_values(ascending=False)[:10].index, :]['Longitude'], 
                      lat=df_merge.iloc[df_merge['Customers'].sort_values(ascending=False)[:10].index, :]['Latitude'], 
                      text=df_merge.iloc[df_merge['Customers'].sort_values(ascending=False)[:10].index, :][['Store_Name', 'Customers']], 
                      name='',
                      marker=dict(size=df_merge.iloc[df_merge['Customers'].sort_values(ascending=False)[:10].index, :]['Customers']/3000, 
                                  color='green'))]

In [ ]:
layout = go.Layout(title=go.layout.Title(text='Top 10 outlets in terms of Number of Customers'), showlegend=False,
                   geo=go.layout.Geo(resolution=110, scope='world', showframe=False, showcoastlines=True,
                                     landcolor="rgb(229, 229, 229)", countrycolor="white", coastlinecolor="white"))

In [ ]:
fig = go.Figure(data=data, layout=layout)

In [ ]:
iplot(fig)

- In case of the **Customer count, 6 outlets** are from the **US** and **4** are from **India** in our **Top 10** outlets with **highest number of customers**.


- The **tooltip** shows the **geographical coordinates, name** and **number of customers** of the **outlet**.

<a id=section502></a>
## 5.2 Analysis Based on Nutritional Value

<a id=section50201></a>
### 5.2.1 Which is the most common category on menu?

In [ ]:
plt.figure(figsize = (15, 6))
df_merge.groupby('Category')['Best_Selling_Item'].count().sort_values(ascending=False).plot(kind='bar', color='g')
plt.ylabel('Count')

- Most items on the **McDonald's menu** belong to the **Hot Beverages** category.

<a id=section50202></a>
### 5.2.2 How many calories (energy) on an average are present in each category of McDonald's menu?

In [ ]:
df_merge.groupby(['Category'])['Energy'].mean().sort_values().plot(kind='barh', figsize=(12,8), fontsize=13, color='red')
plt.xlabel('Calories')

- **Chicken & Fish** category has the **highest calory content** present in them followed by **Sandwiches and Wraps**.


- The **most common category** of **Hot Beverages** have **lesser calory content** than the above mentioned categories.

<a id=section50203></a>
### 5.2.3 Is there any difference in the nutritional value of a grilled vs crispy chicken?

In [ ]:
df_merge['Grilled'] = df_merge['Best_Selling_Item'].str.contains("Grilled")

In [ ]:
crispy_df1 = df_merge.loc[df_merge.Grilled == True, 'Best_Selling_Item'].str.replace('Grilled', 'Crispy')
crispy_df = df_merge.loc[df_merge['Best_Selling_Item'].isin(crispy_df1), ['Best_Selling_Item', 'Total_Fat']]
grilled_df = df_merge.loc[df_merge.Grilled == True, ['Best_Selling_Item', 'Total_Fat', ]]

In [ ]:
df1 = grilled_df.reset_index(drop=True).merge(crispy_df.reset_index(drop=True), left_index=True, right_index=True)
df1.columns = ['Items-Grilled', 'Total_Fat-Grilled', 'Items-Crispy', 'Total_Fat-Crispy']

In [ ]:
df1 = df1.drop('Items-Crispy', axis=1)
df1['Item'] = df1['Items-Grilled'].str.replace("Grilled", "")
df1 = df1.drop('Items-Grilled', axis=1)
df1.index = df1.Item

In [ ]:
df1[['Total_Fat-Grilled', 'Total_Fat-Crispy', 'Item']].plot(kind='barh', title="Fat grilled versus crispy", fontsize=13,
                                                            cmap='viridis', figsize=(10, 8))
plt.xlabel('Total Fat')

- It is evident form the chart that **Crispy** food items have **higher fat content** present as compared to **Grilled** food items.


- These items (Chicken and Sandwiches) already have a **high calory content** as shown in the previous chart, now we can distinguish them on the basis of **Fat content** as well.

<a id=section50204></a>
### 5.2.4 On an average, how much sugar consumption happens when you order Desserts, Beverages, Smoothies and Shakes?

In [ ]:
df_merge.groupby(['Category'])['Sugars'].mean().sort_values().plot(kind='bar', figsize=(15,8), fontsize=13, color='orange')
plt.ylabel('Sugar')

- The **Sugar** content present in the **Desserts, Beverages, Smoothies and Shakes** categories is **more than twice** the **sugar** content present in the remaining categories.


- The **most common category** on the menu **Hot Beverages** is only behind **Smoothies and Shakes** in terms of **Sugar content** present in them.


- Take a **cautious approach** with the intake of the items belonging to these categories because **too much sugar intake** has a lot of **side effects**.

<a id=section50205></a>
### 5.2.5 In what sort of foods and beverages do manufacturers include fiber?

In [ ]:
plt.figure(figsize=(16, 7))
sns.boxplot(data=df_merge, x='Category', y='Dietary_Fibre', palette='viridis', width=0.8)
plt.xticks(rotation=90)

- Not only are salads healthy but also have **high fiber content** compared to the other items on the **McDonald's menu**. But, McDonald's only sell a **few different items** in the **Salads** category.


- The categories **Condiments, Cold Beverages** and **Desserts** have **fiber content** near **zero** with some **outliers**.


- The **Hot Beverages** category also has a **very low fiber content** present in it's items. 

In [ ]:
category_count = df_merge.groupby('Category').count()
item_count = category_count[['Best_Selling_Item']].sort_values('Best_Selling_Item', ascending=False)
item_count

- **30%** of McDonald's menu is comprised of **Hot Beverages** while only **2%** is comprised of **Salads**.


- This shows that **McDonald's doesn't care about your health** that much and **will serve** you **items which aren't good for** your **health** in the long run.

<a id=section50206></a>
### 5.2.6 How to select Nutritious and Non-Nutritious food from menu?

In [ ]:
Nutritious = ['Protein', 'Dietary_Fibre'] 
Non_Nutritious = ['Total_Fat', 'Saturated_Fat', 'Trans_Fat', 'Cholestrol'] 

df_merge['Nutritious'] = df_merge['Protein'] + df_merge['Dietary_Fibre']
df_merge['Non-Nutritious'] = df_merge['Total_Fat'] + df_merge['Saturated_Fat'] + df_merge['Trans_Fat'] + df_merge['Cholestrol']

In [ ]:
df_nutritious = df_merge[df_merge['Category'] == 'Breakfast']
df_nutritious = df_nutritious.groupby(['Best_Selling_Item']).sum().sort_values(by='Nutritious', ascending=False).head(10)

In [ ]:
plt.figure(figsize=(8, 6))
df_nutritious = df_nutritious[Nutritious]
sns.heatmap(df_nutritious, annot=True, cmap='viridis')
plt.title('Food items with high Protein and Fibre Content')

In [ ]:
df_nutritious = df_merge[df_merge['Category'] == 'Breakfast'] 
df_nutritious = df_nutritious.groupby(['Best_Selling_Item']).sum().sort_values(by='Non-Nutritious', ascending=False).head(10)

In [ ]:
plt.figure(figsize=(10, 6))
df_nutritious = df_nutritious[Non_Nutritious]
sns.heatmap(df_nutritious, annot=True, cmap='summer')
plt.title('Food items with Fat and Cholestrol')

- **Big Breakfast with Hotcakes and Egg Whites (Large Biscuit)** contains **high** amount of **Protein** and **Dietary_Fibre** as seen in the **1st chart** and a **lower** content of **Fat** and **Cholestrol** as seen in the **2nd chart**.


- So it can be a preferable food item for people looking for **higher nutrition** and **lower fats and cholestrol**.

<a id=section50207></a>
### 5.2.7 Which is the most proteinaceous item on the menu?

In [ ]:
trace = go.Scatter(y=df_merge['Protein'].values, x=df_merge['Best_Selling_Item'].values, mode='markers',
                   marker=dict(size=df_merge['Protein'].values, color=df_merge['Protein'].values, colorscale='Rainbow',
                               showscale=True))

In [ ]:
data = [trace]

In [ ]:
layout = go.Layout(autosize=True, title = go.layout.Title(text='Scatter plot of Protein per Item on the Menu'), 
                   hovermode='closest', xaxis=dict(showgrid=False, zeroline=False, showline=False), 
                   yaxis=dict(title='Protein', ticklen=5, gridwidth=2, showgrid=False, zeroline=False, showline=False), 
                   showlegend=False)

In [ ]:
fig = go.Figure(data=data, layout=layout)
iplot(fig, filename='scatterChol')

- It's quite evident that the **most proteinaceous item** on the menu is **Chicken McNuggets (40 pieces).**


- The **tooltip** shows the **name** and **protein content** of the **item**.

<a id=section503></a>
## 5.3 Analysis Based on Geographical Information

<a id=section50301></a>
### 5.3.1 Which city has the highest number of McDonald's outlets per capita?

In [ ]:
df_merge['City'].value_counts()[:10].plot(kind='bar', figsize=(15,7), fontsize=13, color='red')
plt.ylabel('Count')

- **Mumbai** is the city with the **highest** number of **outlets** in both **US and India**.

<a id=section50302></a>
### 5.3.1 Which state has the highest number of McDonald's outlets per capita?

In [ ]:
df_merge['State'].value_counts()[:10].plot(kind='barh', figsize=(15,7), fontsize=13, color='black')
plt.xlabel('Count')

- **Maharashtra (MH)** is the state with the **highest** number of **outlets** in both **US and India**, because of the fact that the city with the highest number of stores is **Mumbai** and is the **capital** of **Maharashtra (MH)**.

<a id=section50303></a>
### 5.3.3 Outlet locations in India.

In [ ]:
data = [go.Scattergeo(lon=df_merge[df_merge['Country'] == 'IN']['Longitude'], 
                              lat=df_merge[df_merge['Country'] == 'IN']['Latitude'], 
                              text=df_merge[df_merge['Country'] == 'IN']['Store_Name'], name='',
                              marker=dict(size=df_merge[df_merge['Country'] == 'IN']['Revenue'], color='Blue'))]

In [ ]:
layout = go.Layout(title=go.layout.Title(text='Outlet Locations in India'), showlegend=False,
                   geo=go.layout.Geo(resolution=110, scope='asia', showframe=False, showcoastlines=True,
                                     landcolor="rgb(229, 229, 229)", countrycolor="white", coastlinecolor="white"))

In [ ]:
fig = go.Figure(data=data, layout=layout)

In [ ]:
# It might take some time to load for the first time depending on the system configurations.

fig.add_trace(go.Scattergeo(lon=[77.4126], lat=[23.2599], text=['India'], mode='text', showlegend=False, geo='geo'))

- It can be seen from the above plot that most of the **McDonald's outlets** in **India** are **concentrated** near **big cities** only.


- The **size of the marker** gives the **Revenue generated** by the outlet, **bigger** the **marker higher** the **revenue**.


- The **tooltip** shows the **geographical coordinates** and **name** of the **outlet**.

<a id=section50304></a>
### 5.3.4 Outlet locations in the US

In [ ]:
data = [go.Scattergeo(lon=df_merge[df_merge['Country'] == 'US']['Longitude'], 
                              lat=df_merge[df_merge['Country'] == 'US']['Latitude'], 
                              text=df_merge[df_merge['Country'] == 'US']['Store_Name'], name='',
                              marker=dict(size=df_merge[df_merge['Country'] == 'US']['Revenue']/5, color='Green'))]

In [ ]:
layout = go.Layout(title=go.layout.Title(text='Outlet Locations in the US'), showlegend=False,
                   geo=dict(scope='usa', projection=go.layout.geo.Projection(type='albers usa'), showlakes=True,
                            lakecolor='rgb(255, 255, 255)'))

In [ ]:
fig = go.Figure(data=data, layout=layout)

In [ ]:
iplot(fig, filename='d3-cloropleth-map')

- Unlike **India**, the **outlets** in the **US** are **spread thorughout** the **country** and are not limited to the **big cities**.


- The **size of the marker** gives the **Revenue generated** by the outlet, **bigger** the **marker higher** the **revenue**.


- The **tooltip** shows the **geographical coordinates** and **name** of the **outlet**.

<a id=section504></a>
## 5.4 Analysis Based on Outlet metrics and Nutritional value

<a id=section50401></a>
### 5.4.1 Which outlets have most nutritious item as their best selling item?

In [ ]:
df_merge.iloc[df_merge['Nutritious'].sort_values(ascending=False)[:10].index, :]

In [ ]:
data = [go.Scattergeo(lon=df_merge.iloc[df_merge['Nutritious'].sort_values(ascending=False)[:10].index, :]['Longitude'], 
                      lat=df_merge.iloc[df_merge['Nutritious'].sort_values(ascending=False)[:10].index, :]['Latitude'], 
                      text=df_merge.iloc[df_merge['Nutritious'].sort_values(ascending=False)[:10].index, :][['Store_Name', 'Nutritious']], 
                      name='',
                      marker=dict(size=df_merge.iloc[df_merge['Nutritious'].sort_values(ascending=False)[:10].index, :]['Nutritious']/4, 
                                  color='Red'))]


In [ ]:
layout = go.Layout(title=go.layout.Title(text='Top 10 outlets in term of Nutritional Content of the Best Selling Item'), 
                   showlegend=False,
                   geo=dict(scope='usa', projection=go.layout.geo.Projection(type='albers usa'), showlakes=True,
                            lakecolor='rgb(255, 255, 255)'))

In [ ]:
fig = go.Figure(data=data, layout=layout)

In [ ]:
iplot(fig)

- The **size of the marker** gives the **nutritional content** of the **best selling item** of the outlet, **bigger** the **marker higher** the **nutritional content** in the item.


- The **tooltip** shows the **geographical coordinates** and **name** of the **outlet** along with the **nutritional content** value from the **Nutritious** column.

<a id=section50402></a>
### 5.4.2 What is the Revenue  of the outlet based on the category of its best selling item?

In [ ]:
df_merge.groupby(['Category'])['Revenue'].mean().sort_values().plot(kind='bar', figsize=(15,7), fontsize=13, color='orange')
plt.ylabel('Revenue (in million INR)')

- Outlets with their **Best Selling Item** belonging to the **Salads** category earn the **largest revenue** followed by **Beef & Pork** category.


- This tells us that although the **Salads** category have **lower number of items** on the menu, but still the **outlets** having the items belonging to this category as their **best selling item** are generating **very high revenue** as compared to other categories.


- This might be because of the fact that many people have become **health conscious** and only try to **eat healthy food**.

<a id=section50403></a>
### 5.4.3 How much is the Total Selling Price of the outlet based on the category of its best selling item?

In [ ]:
df_merge.groupby(['Category'])['Selling_Price'].mean().sort_values().plot(kind='bar', figsize=(15,7), fontsize=13, 
                                                                          color='green')
plt.ylabel('Total Selling Price (in million INR)')

- Outlets with their **Best Selling Item** belonging to the **Beef & Pork** category have the **highest total selling price** followed by **Salads** category.

<a id=section50404></a>
### 5.4.4 How much is the Total Cost Price of the outlet based on the category of its best selling item?

In [ ]:
df_merge.groupby(['Category'])['Cost_Price'].mean().sort_values().plot(kind='bar', figsize=(15,7), fontsize=13, 
                                                                       color='violet')
plt.ylabel('Total Cost Price (in million INR)')

- Outlets with their **Best Selling Item** belonging to the **Salads** category have the **highest total cost price** closely followed by **Beef & Pork** category.

<a id=section505></a>
## 5.5 Analysis Based on Outlet Metrics and Geographical Information

<a id=section50501></a>
### 5.5.1 Where did McDonald's enter into Joint Venture to start their Outlets?

In [ ]:
sns.catplot(x='Ownership_Type', y='Number_of_Employees', hue='Country', data=df_merge, height=7, kind='bar', 
            palette='spring')

- McDonald's has **Joint Venture** with other companies in major Asian economies like **India**.


- All the outlets in India are under **Joint Venture between McDonald's, Connaught Plaza Restaurants Limited (CPRL), and Hardcastle Restaurants Pvt. Ltd.** while **American outlets are Company Owned or Licensed**.


- In India, **McDonald's is a 50:50 Joint Venture** company managed by two Indians. 


- While Amit Jatia, M.D. Hardcastle Restaurants Pvt. Ltd. owns and spearheads McDonalds in west & south India, McDonald’s restaurants in North & East India are owned and managed by Vikram Bakshi’s Connaught Plaza Restaurants Private Limited. 

<a id=section50502></a>
### 5.5.2 Top 10 outlets in India and US based on revenue

In [ ]:
top10 = pd.concat([df_merge[df_merge['Country'] == 'IN'].iloc[df_merge[df_merge['Country'] == 'IN']['Revenue'].sort_values(ascending=False)[:10].index, :],
                   df_merge.iloc[df_merge['Revenue'].sort_values(ascending=False)[:10].index, :]])
top10.tail()

In [ ]:
data = [go.Scattergeo(lon=top10['Longitude'], lat=top10['Latitude'], text=top10[['Store_Name', 'Revenue']], name='',
                      marker=dict(size=top10['Revenue']/2.5, color='Red'))]

In [ ]:
layout = go.Layout(title=go.layout.Title(text='Top 10 outlets in India and US based on Revenue (in million INR)'), 
                   showlegend=False,
                   geo=go.layout.Geo(resolution=110, scope='world', showframe=False, showcoastlines=True,
                                     landcolor="rgb(229, 229, 229)", countrycolor="white", coastlinecolor="white"))

In [ ]:
fig = go.Figure(data=data, layout=layout)

In [ ]:
iplot(fig)

- It can be seen through the **size** of the **points** that the **Revenue** of outlets in **India** is **lower** than that in the **US**.


- The **tooltip** shows the **geographical coordinates** and **name** of the **outlet** along with the **Revenue** of the **outlet**.

<a id=section50503></a>
### 5.5.3 What is the mean revenue of outlets in each US state?

In [ ]:
data = [go.Choropleth(colorscale='Viridis', autocolorscale=False, 
                      locations=sorted(df_merge[df_merge['Country'] == 'US']['State'].unique()), locationmode='USA-states', 
                      z=df_merge[df_merge['Country'] == 'US'].groupby(['State'])['Revenue'].mean(), text='Revenue', 
                      colorbar=go.choropleth.ColorBar(title="Revenue (in million INR)"), 
                      marker=go.choropleth.Marker(line=go.choropleth.marker.Line(color='rgb(255,255,255)', width=2)))]

In [ ]:
layout = go.Layout(title=go.layout.Title(text='Mean Revenue of Outlets in each US State (in million INR)'), 
                   geo=dict(scope='usa', projection=go.layout.geo.Projection(type='albers usa'), showlakes=True, 
                            lakecolor='rgb(255, 255, 255)'))

In [ ]:
fig = go.Figure(data=data, layout=layout)

In [ ]:
iplot(fig)

- The **state** with the **highest revenue** in the **US** is **Oklahoma (OK)** with **38.603 million INR** in **Revenue**. 

<a id=section50504></a>
### 5.5.4 How much is the Gross Profit Margin of each outlet?

In [ ]:
data = [go.Scattergeo(lon=df_merge['Longitude'], lat=df_merge['Latitude'], text=df_merge[['Store_Name', 'Gross_Profit_Margin']],
                      name='',
                      marker=dict(size=df_merge['Revenue']/2.5, color='Purple'))]

In [ ]:
layout = go.Layout(title=go.layout.Title(text='Gross Profit Margin of each outlet (in million INR)'), 
                   showlegend=False,
                   geo=go.layout.Geo(resolution=110, scope='world', showframe=False, showcoastlines=True,
                                     landcolor="rgb(229, 229, 229)", countrycolor="white", coastlinecolor="white"))

In [ ]:
fig = go.Figure(data=data, layout=layout)

In [ ]:
iplot(fig)

- It can be seen through the **size** of the **points** that the **Gross Profit Margin** of outlets in **India** is **lower** than that in the **US**.


- The **tooltip** shows the **geographical coordinates** and **name** of the **outlet** along with the **Gross Profit Margin** of the **outlet**.

<a id=section506></a>
## 5.6 Analysis based on Nutritional Value and Geographical Information

<a id=section50601></a>
### 5.6.1 Comparison of nutritional content of each category between India and US

In [ ]:
nutritional_info = ['Energy', 'Protein', 'Sugars', 'Total_Fat', 'Saturated_Fat', 'Cholestrol', 'Carbohydrates', 
                    'Dietary_Fibre', 'Sodium']

In [ ]:
for value in nutritional_info:
    plt.figure(figsize=(15,5))
    df_merge[df_merge['Country'] == 'US'].groupby(['Category'])[value].mean().plot(kind='bar', color='blue', )
    df_merge[df_merge['Country'] == 'IN'].groupby(['Category'])[value].mean().plot(kind='bar', color='red', fontsize=13)
    plt.ylabel(value)
    plt.legend(['USA', 'India'])

From the above charts it is quite evident that **US is rich in nutritional content**. Perhaps, use of **Genetically Modified Food/Crop** might be the reason.


- Some **genetically modified foods** are **designed** to **improve nutrition, quality and taste**. For example, potatoes are modified to even out distribution of starches, enhance texture and reduce fat absorption.


- The **United States leads other countries** in **growing genetically modified foods**. In 2006, 53% of the crops grown in the United States were genetically modified, according to the Human Genome Project. Soybeans, corn and canola are the most common genetically modified crops.

<a id=section50602></a>
### 5.6.2  How are Indian menu items compare to US menu items in terms of nutrition?

In [ ]:
f, axes = plt.subplots(9, 1, figsize=(11, 50))

for key, value in enumerate(nutritional_info):
    sns.boxplot(data=df_merge, x=value, y='Country', orient='h', ax=axes[key], palette='winter')

From the above charts, we can infer the **difference** in the **nutritional content between** the **US** and **Indian McDonald's menu**.

**Scientists** across the world have **identified** two **reasons** for this **declining food nutrition**. 

- One, **intensive agricultural practices** have **stripped the soil of micronutrients**. This could well be the reason for **India** where **soils** have been **found deficient in nutrients**.


- Second, **rising levels of carbon dioxide (CO2) in** the **environment** could also be **affecting plant nutrition levels**. **High CO2 levels in** the **atmosphere lower** the **nitrogen concentration in plants**, which in turn **affects** the **protein content** in food.  

<a id=section507></a>
## 5.7 Analysis based on outlet metrics, menu items, and geographical information

<a id=section50701></a>
### 5.7.1 How is the revenue generation based on different menu items in India and US?

In [ ]:
df_merge.groupby(['Country', 'Category'])['Revenue'].mean().plot(kind='pie', autopct='%3.1f%%', wedgeprops=dict(width=0.15), 
                                                                 explode=np.ones(20)/10, figsize=(13, 12), fontsize=11, 
                                                                 startangle=20, shadow=True, cmap='inferno')

We could infer the following from pie chart above:

- **Maximum revenue generation** for **McDonald's** is **from US** as **compared to India**. 


- The reason behind this could be the fact that **McDonald's** have to enter into **Joint-Ventures** in **India**. 

  - Also the **menu** in **India** is not that **diverse** and only a **limited number of items** are **sold in India** when **compared to** the **menu** in the **US**.
  
  - The **number of outlets** in **India** is **lower** as well when **compared to** the **US**.

<a id=section6></a>
## 6. Conclusion

- It is analyzed that the items in menu dataset can be categorized as **nutritious** food and **non-nutritious** food based on different chart diagrams and range values in percentage obtained. 


- So it is beneficial for demonstrating different range values for food nutrients such as **Protein, Sugar, Dietary Fibers, Fats, Carbohydrates, Cholesterol, and Sodium** for their proper consumption from menu items.


- The US food industry has risen as a high-development and high-benefit area because of its huge potential for esteem expansion, especially inside the food processing industry.


- However, India is still taking its initial steps and this could be the reason for McDonald's India not being profitable after many years of operations. 

<a id=section601></a>
### 6.1 Actionable Insights

- In order to **increase** the **outlet metrics** like **Revenue, Profits** in **Indian outlets**, **McDonald's** need to **open new outlets** belonging to the **Company Owned** and **Licensed** ownership types **instead of Joint Ventures**.


- The **nutritional content** of the food items can be **improved** if **good agricultural practices** are taken up in **India** like use of **Genetically Modified Crops (GM Crops)**, **High-Yield-Variety (HYV) seeds**, etc.


- **McDonald's India** needs to **introduce more food items** on the menu, which have **higher nutritional content** like US and will eventually help them **increase** their **revenue and profits**.